proto/workload.proto

```diff
@@ -58,12 +58,13 @@ message Service {
   // Namespace represents the namespace for the service.
   string namespace = 2;
   // Hostname represents the FQDN of the service.
-  // For Kubernetes, this would be <name>.<namespace>.svc.<cluster domain>
+  // For Kubernetes, this would be <name>.<namespace>.svc.<cluster domain>.
   // TODO: support this field
   string hostname = 3;
   // Address represents the addresses the service can be reached at.
   // There may be multiple addresses for a single service if it resides in multiple networks,
   // multiple clusters, and/or if it's dual stack (TODO: support dual stack).
+  // For a headless kubernetes service, this list will be empty.
```

Comment:
How about cross network

Reply:
I mean if a headless service instance is in another network, do not we need to specify the network gateway address?

Reply:
Still not sure I see what that has to do with Service. To access any workload that resides on another network, you just use the gateway address in the workload.

Reply:
Oh IC. headeless service access will require workload, then when ztunnel would fetch headless service?

Reply:
So on demand is not currently supported for DNS. I raised https://github.com/istio/ztunnel/issues/577